crates/y-sweet/src/server.rs

```diff
@@ -385,7 +385,7 @@ impl Server {
             if let Some(token) = token {
                 let authorization = authenticator
                     .verify_doc_token(token, doc, current_time_epoch_millis())
-                    .map_err(|e| (StatusCode::FORBIDDEN, e))?;
+                    .map_err(|e| (StatusCode::UNAUTHORIZED, e))?;
```

Comment:
is this a case where the test is wrong rather than the implementation? I think if the token is provided which it seems like the guard checks here), but the token is invalid, it's a forbidden rather than unauthorized.

@ellipsis-dev do you know?

Reply:
I think if `verify_doc_token` fails, it's because the token is invalid - which should return a 401 - a failure to _authenticate_. (I hate that the spec calls it `401 Unauthorized` since 403s should be used when _authorization_ fails while 401s should be used when _authentication_ fails. 🤷‍♂️)

Reply:
From MDN docs:

The HTTP `401 Unauthorized` [client error response](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) status code indicates that a request was not successful because it lacks _valid_ authentication credentials for the requested resource.

A `401 Unauthorized` is similar to the [403 Forbidden](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) response, except that a 403 is returned when a request contains valid credentials, but the client does not have permissions to perform a certain action.